src/dict.c

```diff
@@ -966,9 +976,10 @@ static int _dictExpandIfNeeded(dict *d)
      * the number of buckets. */
     if (d->ht[0].used >= d->ht[0].size &&
         (dict_can_resize ||
-         d->ht[0].used/d->ht[0].size > dict_force_resize_ratio))
+         d->ht[0].used/d->ht[0].size > dict_force_resize_ratio) &&
+        dictTypeExpandAllowed(d))
     {
-        return dictExpand(d, d->ht[0].used*2);
+        return dictExpand(d, d->ht[0].used + 1);
```

Comment:
Hi @sundb Actually, before this commit, resizing dict is not expected, new hash table size may `*4` old table size,  oran has been explained https://github.com/redis/redis/pull/7954#issuecomment-725544359

Reply:
@ShooterIT You misunderstood, I meant to change comment "we resize doubling the number of buckets." to "we add 1 to the number of buckets"

Reply:
@sundb `used` is not the number of buckets.
Since first check `used >= size` and we will call _dictNextPower(used+1) in `dictExpand`, new table size will be double of the number of buckets in most cases.

Reply:
@ShooterIT So that's it, thank you.